vispy/ext/_bundled/cassowary/simplex_solver.py

```diff
@@ -0,0 +1,612 @@
+                eplus = SlackVariable(prefix='ep', number=self.slack_counter)
+                eminus = SlackVariable(prefix='em', number=self.slack_counter)
+                expr.set_variable(eplus, -1)
+                expr.set_variable(eminus, 1)
+                self.marker_vars[cn] = eplus
+
+                z_row = self.rows[self.objective]
+                # print("z_row", z_row)
+                sw_coeff = cn.strength * cn.weight
+                # if sw_coeff == 0:
+                #     print("cn ==", cn)
+                #     print("adding ", eplus, "and", eminus, "with sw_coeff", sw_coeff)
+                z_row.set_variable(eplus, sw_coeff)
+                self.note_added_variable(eplus, self.objective)
+                z_row.set_variable(eminus, sw_coeff)
+                self.note_added_variable(eminus, self.objective)
+
+                self.insert_error_var(cn, eminus)
+                self.insert_error_var(cn, eplus)
+
+                if cn.is_stay_constraint:
+                    self.stay_error_vars.append((eplus, eminus))
+                elif cn.is_edit_constraint:
+                    prev_edit_constant = cn.expression.constant
+
+        # print('new_expression returning:', expr)
+        if expr.constant < 0:
+            expr.multiply(-1.0)
+        return expr, eplus, eminus, prev_edit_constant
+
+    def begin_edit(self):
+        assert len(self.edit_var_map) > 0
+        self.infeasible_rows.clear()
+        self.reset_stay_constants()
+        self.edit_variable_stack.append(len(self.edit_var_map))
+
+    def end_edit(self):
+        assert len(self.edit_var_map) > 0
+        self.resolve()
+        self.edit_variable_stack.pop()
+        self.remove_edit_vars_to(self.edit_variable_stack[-1])
+
+    def remove_all_edit_vars(self):
+        self.remove_edit_vars_to(0)
+
+    def remove_edit_vars_to(self, n):
+        try:
+            removals = []
+            for v, cei in self.edit_var_map.items():
+                if cei.index >= n:
+                    removals.append(v)
+
+            for v in removals:
+                self.remove_edit_var(v)
+
+            assert len(self.edit_var_map) == n
+
+        except ConstraintNotFound:
+            raise InternalError('Constraint not found during internal removal')
+
+    def add_stay(self, v, strength=WEAK, weight=1.0):
+        return self.add_constraint(StayConstraint(v, strength, weight))
+
+    def remove_constraint(self, cn):
+        # print("removeConstraint", cn)
+        # print(self)
+        self.needs_solving = True
+        self.reset_stay_constants()
+        z_row = self.rows[self.objective]
+
+        e_vars = self.error_vars.get(cn)
+        # print("e_vars ==", e_vars)
+        if e_vars:
+            for cv in e_vars:
+                try:
+                    z_row.add_expression(self.rows[cv], -cn.weight * cn.strength, self.objective, self)
+                    # print('add expression', self.rows[cv])
+                except KeyError:
```

Comment:
AFAIK, it happens if you try to remove a constraint that hasn't been added (or has already been removed)